dom.bs

```diff
@@ -1419,6 +1419,134 @@ that gave folks all the wrong ideas. <a>Events</a> do not represent or cause act
+
+ <pre class=lang-javascript>
+  function doAmazingness({signal}) {
+    return new Promise((resolve, reject) => {
+      // Begin doing amazingness, and call resolve(result) when done.
+      // But also, watch for signals:
+      signal.addEventListener('abort', () => {
+        // Stop doing amazingness, and:
+        reject(new DOMException('Aborted', 'AbortError'));
+      });
+    });
+  }
+ </pre>
+
+ APIs that require more granular control could extend both {{AbortController}} and {{AbortSignal}}
+ according to their needs.
+</div>
+
+<h3 id=interface-abortcontroller>Interface {{AbortController}}</h3>
+
+<pre class="idl">
+[Constructor(), Exposed=(Window,Worker)]
+interface AbortController {
+  [SameObject] readonly attribute AbortSignal signal;
+
+  void abort();
+};
+</pre>
+<dl class=domintro>
+ <dt><code><var>controller</var> = new <a constructor lt="AbortController()">AbortController</a>()</code>
+ <dd>Returns a new <var>controller</var> whose {{AbortController/signal}} is set to a newly
+ created {{AbortSignal}} object.
+
+ <dt><code><var>controller</var> . </code>{{AbortController/signal}}
+ <dd>Returns the {{AbortSignal}} object associated with this object.
+
+ <dt><code><var>controller</var> . </code>{{AbortController/abort()}}
+ <dd>Invoking this method will set this object's {{AbortSignal}}'s [=AbortSignal/aborted flag=],
+ thereby signaling to any observers that the associated activity should be aborted.
+ </dl>
+
+The <dfn attribute for=AbortController><code>signal</code></dfn> attribute must return the value to
+which it was initialized. When a {{AbortController}} is created, the attribute must be initialized
+to a newly created {{AbortSignal}} object.
+
+The <dfn method for=AbortController><code>abort()</code></dfn> method, when invoked, must run
+these steps:
+
+<ol>
+ <li>Let <var>signal</var> be this object's {{AbortController/signal}}.
+ <li>Set <var>signal</var>'s [=AbortSignal/aborted flag=].
+ <li>
+  <p>For each <var>algorithm</var> in <var>signal</var>'s
+  [=AbortSignal/abort callbacks=]:</p>
+
+  <ol>
+   <li>Run <var>algorithm</var>.
+  </ol>
+ </li>
+ <li>[=Fire an event=] named <code>abort</code> at <var>signal</var>.
+</ol>
+
+<h4 id=interface-AbortSignal>Interface {{AbortSignal}}</h4>
+
+<pre class="idl">
+[Exposed=(Window,Worker)]
+interface AbortSignal : EventTarget {
+  readonly attribute boolean aborted;
+
+  attribute EventHandler onabort;
+};
+</pre>
+<dl class=domintro>
+ <dt><code><var>signal</var> . </code>{{AbortSignal/aborted}}
+ <dd>Returns true if this {{AbortSignal}} has been aborted, and false otherwise.
+</dl>
+
+Each {{AbortSignal}} has an <dfn for=AbortSignal>aborted flag</dfn> which is unset unless otherwise
+specified.
+
+Each {{AbortSignal}} has <dfn for=AbortSignal>abort callbacks</dfn>, which is an [=ordered set=] of
+algorithms which are to be executed when its [=AbortSignal/aborted flag=] is set. Unless otherwise
+specified, its value is the empty set.
```

Comment:
Can we name this something different from callbacks. Callback already has a very specific meaning in IDL.